static/js/main.js

```diff
@@ -142,36 +161,45 @@ function init() {
                 sent: false,
                 sender: 'volunteer',
                 from: 'copilot',
+                time: Date.now()
               }, () => {
                 $('#mainInput').val('');
+                updateTimestamps();
               });
           }
         });
 
-        // Notes real-time editing
+        // notes real-time editing
         var typingTimer;
-        $('.notes').keyup(function() {
+        $('.notes').keyup(() => {
             clearTimeout(typingTimer);
-            typingTimer = setTimeout(function() {
+            typingTimer = setTimeout(() => {
               // upload to Firebase
               db.child('cases').child(CURRENT_CASE).child('notes').set($('.notes').val())
             }, 200);
         });
 
         //on keydown, clear the countdown
-        $('.notes').keydown(function() {
+        $('.notes').keydown(() => {
             clearTimeout(typingTimer);
         });
       });
 }
 
-// When logout button is pressed
+// when logout button is pressed
 $("#logOut").click(() => {
   firebase.auth().signOut();
 });
 
-// update bottom of .messageSpace
+// update bottom of .messageSpace UI handler
 function updateScroll() {
   const element = document.getElementsByClassName('messages');
   element[0].scrollTop = element[0].scrollHeight + 100;
 }
+
+// refresh pretty message timestamps
+function updateTimestamps() {
```

Comment:
Time ago is doing it automatically. I have this manual refresh so that each time we load a new set of messages it only activates listeners for the datetimes currently on that page.